src/box/lua/config/applier/credentials.lua

```diff
@@ -653,6 +653,33 @@ local function sync_privileges(credentials, obj_to_sync)
     end)
 end
 
+local schema_is_upgraded = false
+local schema_is_upgraded_cond
+
+local function update_schema_upgraded_status(version)
+    local schema_version = version or mkversion.get()
+    schema_is_upgraded = schema_version == mkversion.get_latest()
+
+    if schema_is_upgraded then
+        schema_is_upgraded_cond:signal()
+    end
+end
+
+local function credentials_upgrade_on_replace(old, new)
+    if old == nil or new == nil then
+        return
+    end
+
+    local latest_version = mkversion.get_latest()
+    local new_version = mkversion.from_tuple(new)
+
+    if new_version == latest_version then
+        box.on_commit(function()
+            update_schema_upgraded_status(new_version)
+        end)
+    end
```

Comment:
Just curious: what is the behavior after `box.schema.downgrade()`?

I think that we can fix upgrading within this patch and postpone a work on downgrading if it requires an extra work, not just a couple of fixups.

I propose to experiment with downgrading a bit and dedice then.

Reply:
Right now after downgrading there is no new behaviour. 

Restarting the tarantool after downgrading results with the message similar to this:
```
2024-08-12 16:39:00.462 [16653] main/118/sync_credentials/tarantool.config log.lua:130 W> credentials: the schema has an old version and user/roles can not be loaded. Consider executing box.schema.upgrade() to perform an upgrade.
```

Reply:
I guess that the following scenario will result in failure:

* A normal (upgraded) instance is run.
* `box.schema.downgrade()` (plus `box.snapshot()`) is called.
* Something changed in the `credentials` section of the config: a new user is added, a role is added, a password is changed or a set of priviliges is changed.
* `config:reload()` is called.

I guess that it results in `check_errors` status in `box.info.config` and an error regarding non-upgraded schema in `box.info.config.alerts`.

I propose to elaborate a solution as part of a separate issue. I suggest to create the issue and mention it in the commit message here to make it clear what is solved by this commit and what doesn't.

Reply:
Thank you for writing these concerns here. I'll create the issue right after I'll be able to reproduce and get these error in order to be sure how the problem looks like.

Reply:
For the record: #10411.